### PR TITLE
Fix Multithreaded Readers working with Unity Catalog on Databricks [databricks]

### PR DESCRIPTION
**Pull request by tgravescs**
Part of https://github.com/NVIDIA/spark-rapids/issues/8210

The issue here is our multi-threaded and coalescing readers fail to read with - org.apache.spark.util.TaskCompletionListenerException: org.apache.spark.SparkException: Missing Credential Scope.

This happens when the Unity Catalog feature on databricks is enabled.  Interestingly enough, its just on at the account level even those this particular job isn't using it directly.

This PR only handles Parquet files because that is all I'm currently able to test with.  I am working on setting up our own environment to test Unity Catalog with but at the moment I tested this fix in customers Databricks 11.3 environment.  I tested multithreaded, coalescing, and perfile. For coalescing I also tested with filtering in parallel on and off.

the only way I've been able to reproduce at this point is to do a delta write to one table and then a read from another one. I have limited options as its customer env. My guess might be that the write is setting up credentials scope to one thing and then when we go to do the read the scope is set wrong and we have to explicitly set it for the read to work.

For the fix, I found a way to get the hadoop conf that seems to have the necessary confs for credentials and you use that conf in the reader threads.

### CODE REVIEW

**Comment by tgravescs** [comment]
need to fix 3.4

**Comment by tgravescs** [comment]
build

**Review by jlowe — DISMISSED** [review]

File: sql-plugin/src/main/spark330db/scala/com/nvidia/spark/rapids/shims/ReaderUtils.scala [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+   */
+  def getHadoopConfForReaderThread(filePath: Path, conf: Configuration): Configuration = {
+    if (isUnityCatalogEnabled) {
+      try {
+        com.databricks.unity.ClusterDefaultSAM.createDelegateHadoopConf(filePath, conf)
+      } catch {
+        case a: AssertionError =>
+          // ignore this and just return the regular conf, it might be a filesystem not supported
+          // and I don't have a good way to check this
+          logWarning("Assertion error calling createDelegeHadoopConf, skipping.", a)
```
Comment (abellina) [inline sql-plugin/src/main/spark330db/scala/com/nvidia/spark/rapids/shims/ReaderUtils.scala:46]:
```suggestion
          logWarning("Assertion error calling createDelegateHadoopConf, skipping.", a)
```

**Comment by tgravescs** [comment]
failure fetching a jar, going to rekick tests

**Comment by tgravescs** [comment]
build

**Review by abellina — APPROVED** [review]

**Pull request merged**